firehose-source/src/main/java/source/ByteBufferMessageConverter.java

```
@@ -0,0 +1,73 @@
+ *
+ */
+
+package source;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import org.cloudfoundry.dropsonde.events.EventFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.converter.AbstractMessageConverter;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MimeTypeUtils;
+import org.springframework.xd.tuple.Tuple;
+import org.springframework.xd.tuple.TupleBuilder;
+import org.springframework.xd.tuple.TupleToJsonStringConverter;
+
+import java.nio.ByteBuffer;
+
+/**
+ * @author Vinicius Carvalho
+ */
+@Component
+public class ByteBufferMessageConverter extends AbstractMessageConverter {
+
+    protected ByteBufferMessageConverter() {
+        super(MimeTypeUtils.APPLICATION_OCTET_STREAM);
+    }
+
+    private TupleToJsonStringConverter jsonConverter = new TupleToJsonStringConverter();
+
+    @Autowired
+    private FirehoseProperties metadata;
+
+    @Override
+    protected boolean supports(Class<?> clazz) {
+        return ByteBuffer.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public Object convertFromInternal(Message<?> message, Class<?> aClass) {
+        ByteBuffer buffer = (ByteBuffer) message.getPayload();
+        byte[] bytes = new byte[buffer.remaining()];
+        buffer.get(bytes, 0, bytes.length);
+        Tuple result = TupleBuilder.tuple().build();
+        try {
+            EventFactory.Envelope envelope = EventFactory.Envelope.parseFrom(bytes);
+            result = TupleFactory.createTuple(envelope);
+        } catch (InvalidProtocolBufferException e) {
+            e.printStackTrace();
+        }
+        return (metadata.isOutputJson()) ? jsonConverter.convert(result) : result;
+    }
+
+    @Override
+    public Object convertToInternal(Object payload, MessageHeaders messageHeaders) {
+        return payload;
```

This feels like it violates symmetry, is this intended? maybe throw an exception here if this method is not meant to be called?